Event journal cleanup

mmedenjak opened the pull request:
- move com.hazelcast.journal to com.hazelcast.internal.journal as the
event journal is internal and there is no public API yet
- unify the event journal tests to use a style similar to the near cache
unified tests
- add bouncing tests for map and cache

Fixes: https://github.com/hazelcast/hazelcast/issues/12152
EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/1928

pveentjer commented on hazelcast/src/main/java/com/hazelcast/cache/impl/journal/DeserializingEventJournalCacheEvent.java:
```diff
@@ -95,4 +95,18 @@ private Data toData(Data data, Object o) {
         serializationService = ((SerializationServiceSupport) hazelcastInstance).getSerializationService();
     }
+
+    @Override
```
Is it not sufficient to add this to the documentation of the class? Why create additional methods that don't do anything?

mmedenjak replied:
Checkstyle and/or findbugs. I'll try find a supression.

mmedenjak replied:
Found the suppression:
```
@SuppressFBWarnings(value = "EQ_DOESNT_OVERRIDE_EQUALS",
        justification = "equality is checked by serialised data in superclass, not deserialised instances in this class")
```
Removed the dummy implementations.

pveentjer commented on hazelcast/src/test/java/com/hazelcast/cache/impl/journal/EventJournalCacheDataStructureAdapter.java:
```diff
@@ -0,0 +1,109 @@
+            long startSequence, int minSize, int maxSize, int partitionId,
+            com.hazelcast.util.function.Predicate<? super EventJournalCacheEvent<K, V>> predicate,
+            Projection<? super EventJournalCacheEvent<K, V>, ? extends T> projection) {
+        return ((EventJournalReader<EventJournalCacheEvent<K, V>>) cache)
```
My glasses just cracked :)

Could you do a argument per method? Then at least it is clear how many arguments and what their position is.

mmedenjak replied:
I haven't split it into multiple methods but I've tried to make it look nicer. Hopefully it works. I can split it into multiple methods if necessary.

pveentjer commented on hazelcast/src/test/java/com/hazelcast/cache/impl/journal/EventJournalCacheDataStructureAdapter.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    private Set<Map.Entry<K, V>> getEntries(ICache<K, V> cache) {
+        final Iterator<Cache.Entry<K, V>> it = cache.iterator();
```
I'll finalize you

mmedenjak replied:
Let's not go into this discussion :)

pveentjer commented on hazelcast/src/test/java/com/hazelcast/journal/EventJournalEventAdapter.java:
```diff
@@ -0,0 +1,57 @@
+ * @param <V>       the event value type
+ * @param <EJ_TYPE> the event type
+ */
+public interface EventJournalEventAdapter<K, V, EJ_TYPE> extends Serializable {
```
Since this class already contains 'event' is it needed to repeat it in every method?

mmedenjak replied:
Fixed.

mmedenjak commented:
@pveentjer thanks for the review! Hope you `finally` find some love for `final` some day :)

Donnerbart commented on hazelcast/src/test/java/com/hazelcast/journal/NewValueIncrementingProjection.java:
```diff
@@ -0,0 +1,40 @@
+    public Integer transform(EJ_TYPE input) {
+        return journalEventAdapter.getNewValue(input) + delta;
+    }
+}
```
Missing newline at EOF

Donnerbart commented on hazelcast/src/test/java/com/hazelcast/journal/NewValueParityPredicate.java:
```diff
@@ -0,0 +1,42 @@
+    public boolean test(EJ_TYPE e) {
+        return journalEventAdapter.getNewValue(e) % 2 == remainder;
+    }
+}
```
Missing newline at EOF

Donnerbart commented on hazelcast/src/test/java/com/hazelcast/journal/TruePredicate.java:
```diff
@@ -0,0 +1,17 @@
+package com.hazelcast.journal;
```
Missing copyright header

Donnerbart commented on hazelcast/src/test/java/com/hazelcast/journal/TruePredicate.java:
```diff
@@ -0,0 +1,17 @@
+    public boolean test(T t) {
+        return true;
+    }
+}
```
Missing newline at EOF

mmedenjak commented:
@Donnerbart thanks for the review, addressed all comments. I realise we can go further with this, adding methods on the `com.hazelcast.internal.adapter.DataStructureAdapter` but I was hesitant to do so for now since it's production code and it's related to the near cache.
I think this is a good step and now it's easier to write journal tests. We can further improve it in a separate PR IMHO.
